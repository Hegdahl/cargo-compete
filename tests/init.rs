pub mod common;

use duct::cmd;
use ignore::overrides::Override;
use insta::{assert_json_snapshot, assert_snapshot};
use std::str;

#[test]
fn no_crate() -> anyhow::Result<()> {
    let (output, tree) = run("\n1\n")?;
    assert_snapshot!("no_crate_output", output);
    assert_json_snapshot!("no_crate_file_tree", tree, { r#".**["Cargo.lock"]"# => ".." });
    Ok(())
}

#[test]
fn use_crate() -> anyhow::Result<()> {
    let (output, tree) = run("\n2\n")?;
    assert_snapshot!("use_crate_output", output);
    assert_json_snapshot!("use_crate_file_tree", tree, { r#".**["Cargo.lock"]"# => ".." });
    Ok(())
}

#[test]
fn use_crate_via_bianry() -> anyhow::Result<()> {
    let (output, tree) = run("\n3\n")?;
    assert_snapshot!("use_crate_via_bianry_output", output);
    assert_json_snapshot!("use_crate_via_bianry_file_tree", tree, { r#".**["Cargo.lock"]"# => ".." });
    Ok(())
}

fn run(input: &'static str) -> anyhow::Result<(String, serde_json::Value)> {
    common::run(
        |workspace_root| -> _ {
            println!("{}", cmd!("git", "init", workspace_root).read()?);
            Ok(())
        },
        input.as_bytes(),
        &["", "compete", "i"],
        |workspace_root, output| {
            output
                .replace(workspace_root.to_str().unwrap(), "{{ cwd }}")
                .replace(std::path::MAIN_SEPARATOR, "{{ main_path_separator }}")
        },
        |_| Ok(Override::empty()),
    )
}
