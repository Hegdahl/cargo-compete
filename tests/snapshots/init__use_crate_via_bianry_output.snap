---
source: tests/init.rs
expression: output
---
Websites you compete in:
1 AtCoder
2 Codeforces
3 yukicoder
Space-delimited numbers (defaults to all): Do you use crates on AtCoder?
1 No
2 Yes
3 Yes, but I submit base64-encoded programs
Number:        Wrote {{ cwd_canonicalized }}.gitignore
       Wrote {{ cwd_canonicalized }}atcoder{{ main_path_separator }}Cargo.toml
       Wrote {{ cwd_canonicalized }}atcoder{{ main_path_separator }}compete.toml
     Created `cargo-compete-template` package at {{ cwd_canonicalized }}atcoder{{ main_path_separator }}cargo-compete-template
    Updating {{ cwd_canonicalized }}atcoder{{ main_path_separator }}Cargo.lock
       Wrote {{ cwd_canonicalized }}codeforces{{ main_path_separator }}Cargo.toml
       Wrote {{ cwd_canonicalized }}codeforces{{ main_path_separator }}rust-toolchain
       Wrote {{ cwd_canonicalized }}codeforces{{ main_path_separator }}compete.toml
     Created `cargo-compete-template` package at {{ cwd_canonicalized }}codeforces{{ main_path_separator }}cargo-compete-template
    Updating {{ cwd_canonicalized }}codeforces{{ main_path_separator }}Cargo.lock
       Wrote {{ cwd_canonicalized }}yukicoder{{ main_path_separator }}Cargo.toml
       Wrote {{ cwd_canonicalized }}yukicoder{{ main_path_separator }}rust-toolchain
       Wrote {{ cwd_canonicalized }}yukicoder{{ main_path_separator }}compete.toml
     Created `cargo-compete-template` package at {{ cwd_canonicalized }}yukicoder{{ main_path_separator }}cargo-compete-template
    Updating {{ cwd_canonicalized }}yukicoder{{ main_path_separator }}Cargo.lock

